lib/rules/no-return-await.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * @fileoverview Disallows unnecessary `return await`
+ * @author Jordan Harband
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow unnecessary `return await`",
+            category: "Best Practices",
+            recommended: false // TODO: set to true
+        },
+        fixable: false,
+        schema: [
+        ]
+    },
+
+    create(context) {
+        return {
+            ReturnStatement(node) {
+                if (node.argument && node.argument.type === "AwaitExpression") {
+                    context.report({
+                        node,
+                        loc: node.argument.loc,
+                        message: "`return await` from an `async function` is redundant - just `return` the value directly.",
+                    });
+                }
+            },
+
+        };
+    }
```

Comment:
I'm having trouble figuring out how to get at an implicit return expression in an arrow function - any pointers?